roles/db/tasks/main.yml

```diff
@@ -1,32 +1,165 @@
+    dest: "{{pgconfdir}}/server.key"
+    owner: postgres
+    group: postgres
+    mode: 0400
+    remote_src: True
+    follow: True
+  notify: Restart postgres
+
+- name: Add postgres prehook for certbot renewal
+  blockinfile:
+    create: True
+    dest: "/etc/letsencrypt/hooks/pre.sh"
+    mode: 0755
+    marker: "# {mark} ANSIBLE MANAGED BLOCK postgres pre hook"
+    block: |
+      systemctl stop {{postgres_service}}
+
+- name: Add postgres posthook for certbot renewal
+  blockinfile:
+    create: True
+    dest: "/etc/letsencrypt/hooks/post.sh"
+    mode: 0755
+    marker: "# {mark} ANSIBLE MANAGED BLOCK postgres post hook"
+    block: |
+      cp {{lets_encrypt_priv_key}} {{pgconfdir}}/server.key
+      cp {{lets_encrypt_pub_key}} {{pgconfdir}}/server.crt
+      systemctl start {{postgres_service}}
+
+- name: Open firewalld for postgres
+  firewalld:
+    rich_rule: "rule family=\"ipv4\" service name=\"postgresql\" source address=\"{{hostvars[item]['ansible_default_ipv4']['address']}}\" accept"
+    state: enabled
+    permanent: true
+    immediate: true
+  with_items: "{{groups['app']}}"
+
 - name: Start/enable Postgres
-  service: name=postgresql state=started enabled=yes
+  service:
+    name: "{{postgres_service}}"
+    state: started
+    enabled: yes
 
 - name: Create database user
-  postgresql_user: name={{db_user}} password={{db_pass}} login_unix_socket=/var/run/postgresql/
+  postgresql_user: 
+    name: "{{db_user}}"
+    password: "{{db_pass}}"
+    login_unix_socket: /var/run/postgresql/
   become: true
   become_user: postgres
 
 - name: Create database
-  postgresql_db: name={{db_name}} owner={{db_user}} login_unix_socket=/var/run/postgresql/
+  postgresql_db:
+    name: "{{db_name}}"
+    owner: "{{db_user}}"
+    login_unix_socket: /var/run/postgresql/
   become: true
   become_user: postgres
+
+- name: Create DB backup directory
+  file: 
+    path: "{{pgdata}}/backups"
+    state: directory
+    owner: postgres
+    group: postgres
+    mode: 0755
+
+- name: Install awscli for S3 backups
+  package:
+    name: awscli
+    update_cache: yes
+
+- name: Copy AWS S3 credentials
+  lineinfile:
+    dest: "{{'~postgres'|expanduser}}/.bashrc"
+    line: "export {{item.variable}}=\"{{item.value}}\""
+    regexp: "^export {{item.variable}}="
+    create: true
+  with_items:
+    - { variable: 'AWS_ACCESS_KEY_ID', value: "{{s3_aws_access_key}}" }
+    - { variable: 'AWS_SECRET_ACCESS_KEY', value: "{{s3_aws_secret_key}}"}
+  
+- name: Create DB backup job
+  cron:
+    name: "db backup" 
+    cron_file: db_backup
+    special_time: daily
+    user: postgres
+    state: present
+    job: "/usr/bin/pg_dump {{db_name}}  | xz >  {{pgdata}}/backups/{{db_name}}_db_{{ansible_fqdn}}_$(date +\\%Y\\%m\\%d).sql.xz && aws s3 cp {{pgdata}}/backups/{{db_name}}_db_{{ansible_fqdn}}_$(date +\\%Y\\%m\\%d).sql.xz s3://{{s3_bucket_name}}/{{ansible_fqdn}}/{{db_name}}_db_{{ansible_fqdn}}_$(date +\\%Y\\%m\\%d).sql.xz"
```

Comment:
I kind of like the idea of periodically running aws s3 sync of the directory rather than aws s3 cp.  That way, if one cp fails, the next sync will copy the failed one and the new one both.